src/compiler/utilities.ts

```diff
@@ -10518,7 +10517,7 @@ export function canHaveExportModifier(node: Node): node is Extract<HasModifiers,
 }
 
 /** @internal */
-export function isOptionalJSDocPropertyLikeTag(node: Node): node is JSDocPropertyLikeTag {
+export function isOptionalJSDocPropertyLikeTag(node: Node): boolean {

```

Comment:
If I'm not mistaken this function wasn't proving that the input is `JSDocPropertyLikeTag` - some extra checks on a variable with `JSDocPropertyLikeTag` type were done here and it could still return `false` for it. So this wasn't a correct return type annotation

Reply:
It's certainly not the first type guard we've had which has lied for convenience's sake. But it doesn't seem like it really helped here.